src/includes/traits/Plugin/AutoCacheUtils.php

```diff
@@ -170,5 +170,49 @@ public function autoCacheCheckXmlSitemap($sitemap, $is_nested_sitemap = false, $
 
         return true;
     }
+
+    /**
+     * Download XML Sitemap via cURL and store in a temp file.
+     *
+     * @since 16xxxx
+     *
+     * @param string $url to XML Sitemap that needs to be downloaded
+     *
+     * @throws \Exception If unable to write temporary file or download XML Sitemap
+     *
+     * @return string Path to downloaded XML Sitemap file
+     */
+    public function autoCacheGetSitemapViaCurl($url)
+    {
+        if (!($tmp_dir = $this->getTmpDir())) {
+            throw new \Exception(__('No writable tmp directory.', SLUG_TD));
+        }
+
+        $tmp_file = $tmp_dir.'/'.$this->addTmpSuffix($tmp_dir).'.xml';
```

Comment:
Hmm, the PHPDoc Block for `addTmpSuffix()` should be updated in that case. It currently implies that it can be used for a file path:

```php
/**
 * Adds a tmp name suffix to a directory/file path.
 * ...
*/
```

Looking over the source for `addTmpSuffix()` tells me that it works just as I would expect, so I'm not sure why you say it's not what I'd expect.